.circleci/config.yml

```diff
@@ -81,27 +57,44 @@ jobs:
 
 workflows:
   version: 2
-  build_n_deploy:
+
+  pull_request:
     jobs:
-      - "build_docker":
-          <<: *transient
       - "test":
-          <<: *transient
-          requires:
-            - "build_docker"
+          filters:
+            tags:
+              ignore: /.*/
+            branches:
+              ignore:
+                - "dev"
+                - "master"
+
+  release_dev:
+    jobs:
+      - "test":
+          filters:
+            tags:
+              ignore: /.*/
+            branches:
+              only: dev
       - "publish_docker_dev":
-          <<: *onlyBranchDev
           requires:
             - "test"
       - "release_cli_dev":
-          <<: *onlyBranchDev
           requires:
             - "publish_docker_dev"
+
+  release_prod:
+    jobs:
+      - "test":
+          filters:
+            tags:
+              only: /^v.*/
+            branches:
+              only: master
```

Comment:
The tests here will be executed only when we are on master AND we have a tag. In this case we skip all the tests when we merge to master and these are important.

We should add the check on the tag only for the `publish_docker_prod` and `release_cli_prod` not for the test